File: comparison_Python_R_surfaceplot_gaussian.r
#!/usr/bin/python
#####################################################################
# Author:		Alberto Albz Marocchino
# Date:			19-02-2017
# Purpose:  comparison between python and R :: contour plot
# Source:   R
#####################################################################

sigma <- matrix(c(3,1,1,1), nrow=2)
data.grid <- expand.grid(x = seq(-5, 5, length.out=100), y = seq(-5, 5, length.out=100))
z <- cbind(data.grid, prob = mvtnorm::dmvnorm(data.grid, mean = c(0,0), sigma = sigma))
ggplot(z, aes(x=x, y=y)) +
    geom_raster(aes(fill = prob)) +
    coord_fixed(xlim = c(-5, 5), ylim = c(-5, 5), ratio = 1)

File: comparison_Python_R_surfaceplot_gaussian.py
#!/usr/bin/python
#####################################################################
# Author:		Alberto Albz Marocchino
# Date:			26-02-2017
# Purpose:     comparison between python and R :: surface plot multivaraite gaussian
# Source:       python
#####################################################################

import numpy as np
import pylab as pyl

#--- generate mesh ---#
x=np.linspace(-5,5,100)
y=np.linspace(-5,5,100)
xx, yy = np.meshgrid(x, y)
z = np.exp(-xx**2/6 - yy**2/2 + xx*yy/2)

%matplotlib inline
plt.contourf(x,y,z,100)
